pyuvdata/uvdata/mir.py

```diff
@@ -274,49 +417,104 @@ def read_mir(
         # telescope geocentric position is required , i.e we are going from
         # relRotECEF -> relECEF
         self.antenna_positions = uvutils.ECEF_from_rotECEF(antXYZ, lon)
-        self.baseline_array = self.antnums_to_baseline(
-            self.ant_1_array, self.ant_2_array, attempt256=False
-        )
 
         self.history = "Raw Data"
         self.instrument = "SWARM"
 
-        # We can just skip an appropriate number of records
-        self.integration_time = mir_data.sp_data["integ"][:: len(corrchunk)].astype(
-            float
-        )
+        # Let's try to chalk up all the per-sp stuff here, eh?
+        sp_to_blt = ["igq", "ipq", "flags", "vradcat"]
+        sp_temp_dict = {}
+        suppress_warning = False
+        for sp_rec in mir_data.sp_data:
+            temp_dict = {item: sp_rec[item] for item in sp_to_blt}
+            try:
```

Comment:
@kartographer -- Can you add a short explanation here for which `dict` throws the keyerror? Lines 440/442 make this a bit confusing.

Reply:
I added some additional comments in 5c36be1a. Basically, what's happening here is that I'm scanning through per-spectral window data to see if things are changing on a per-baseline-time basis. The `dict` basically serves as a record for what baseline-times have been checked -- if no key matches, it means that particular baseline-time has not been evaluated before, hence why I plug in the key after the `except`.